bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtocol.java

```diff
@@ -481,12 +482,14 @@ public ReferenceCounted retain(int increment) {
 
         @Override
         public ReferenceCounted touch() {
-            return data.touch();
```

Comment:
When using the leak checker, this was causing the `ReadResponse` to get converted into an instrumented buffer, and the serialization was completely skipped.